test/IoC.Test/UnitTest/IoCTest.cs

```diff
@@ -0,0 +1,42 @@
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IoC.Test.UnitTest;
+
+public class IoCTest
+{
+    private readonly InstancePool _sut; //SUT stands for System Under Test.
+
+    public IoCTest(ITestOutputHelper output)
+    {
+        _sut = InstancePool.Instance;
+    }
+
+    [Fact]
+    public void ShouldGetARegisteredInstance()
+    {
+        //Given
+        _sut.Register<TestClass>();
+
+        //When
+        var tc = _sut.GetInstance<TestClass>();
+
+        //Then
+        Assert.NotNull(tc);
+    }
+
+
+    [Fact]
+    public void ShouldReturnTheSameGUID()
+    {
+        //Given
+
+        //When
+        var instance1 = _sut.GetInstance<TestClass>();
```

Comment:
Can we compare reference? I think the message above can be address in this test

Reply:
That's correct. The test returns `true` even making several references of the same instance. I got the same Guide ;)